configure.ac

```diff
@@ -530,15 +534,21 @@ AS_IF(
 
 AS_CASE([$ocaml_cv_cc_vendor],
   [xlc-*],
-    [outputobj='-o $(EMPTY)'; gcc_warnings="-qflag=i:i"], # all warnings enabled
+    [outputobj='-o $(EMPTY)'
+    warn_error_flag=''
+    cc_warnings='-qflag=i:i'], # all warnings enabled
   [msvc-*],
-    [outputobj=-Fo; gcc_warnings=""],
-  [outputobj='-o $(EMPTY)'; AS_CASE([AC_PACKAGE_VERSION],
-    [*+dev*],
-      [gcc_warnings="-Wall -Werror"],
-    [gcc_warnings="-Wall"]
-  )]
-)
+    [outputobj='-Fo'
+    warn_error_flag='-WX'
+    cc_warnings=''],
+  [outputobj='-o $(EMPTY)'
+  # FIXME Should also be -Wdeclaration-after-statement here
+  warn_error_flag='-Werror'
+  cc_warnings='-Wall'])
+
+AS_CASE([$enable_warn_error,AC_PACKAGE_VERSION],
+  [yes,*|,*+dev*|*+multicore*],
```

Comment:
I might be getting the wrong end of the stick, but if someone does `--disable-warn-error` explicitly with a `+multicore` version. Will this still force a warn-error build?

That is should it be `,*+multicore*` or `*+multicore*` in the condition

Reply:
You have completely the right end of the stick - there's a comma missing 😊